velox/functions/prestosql/aggregates/AverageAggregate.cpp

```diff
@@ -134,7 +143,12 @@ class AverageAggregate : public exec::Aggregate {
       if (!decodedRaw_.isNullAt(0)) {
         const TInput value = decodedRaw_.valueAt<TInput>(0);
         const auto numRows = rows.countSelected();
-        updateNonNullValue(group, numRows, TAccumulator(value) * numRows);
+        if (aggregateType == AggregateType::kAvg) {
```

Comment:
This is a runtime conditional statement. We avoid conditionals in loops in Velox to avoid branch misses.

This code should be made a compile time check by using constexpr instead.

if constexpr (aggregateType == AggregateType::kAvg)

Please change this in all the usages in this code and then I'll take another look.